scatter-gather/scatter-gather.bal

```diff
@@ -0,0 +1,50 @@
+import ballerina/http;
+
+service /api/v1 on new http:Listener(8080) {
+    
+    resource function get vehicles(string dropOffDateTime, string dropOffLocation, string pickupDateTime, string pickupLocation) returns json|error {
+        return check requestRentalCar(dropOffDateTime, dropOffLocation, pickupDateTime, pickupLocation);
+    }
+}
+
+function transformResponse(map<anydata|error> responses) returns json|error {
+    json[] results = [];
+    foreach var [_, value] in responses.entries() {
+        if value is json {
+            foreach json item in <json[]>(check value.result) {
+                results.push({
+                    carType: check item.carType,
+                    price: check item.price,
+                    resultLink: check item.resultLink
+                });
+            }
+        } else if value is xml {
+            foreach xml item in value/<Response>/<Car> {
+                results.push({
+                    carType: (item/<CarType>).data(),
+                    price: (item/<Price>).data(),
+                    resultLink: (item/<ResultLink>).data()
+                });
+            }
+        }
+    }
+    return results;
+}
+
+function requestRentalCar(string dropOffDateTime, string dropOffLocation, string pickupDateTime, string pickupLocation) returns json|error {
```

Comment:
Shall we move this function to immediate below of the `requestRentalCar` function